src/storage/v2/disk/storage.cpp

```diff
@@ -321,31 +321,29 @@ std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToMa
 }
 
 std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToLabelIndexCache(
-    const rocksdb::Slice &key, const rocksdb::Slice &value, Delta *index_delta,
+    std::string &&key, std::string &&value, Delta *index_delta,
     utils::SkipList<storage::Vertex>::Accessor index_accessor) {
-  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromLabelIndexStorage(key.ToString())));
+  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromLabelIndexStorage(key)));
   if (VertexExistsInCache(index_accessor, gid)) {
     return std::nullopt;
   }
 
-  const std::string value_str{value.ToString()};
-  const auto labels{utils::DeserializeLabelsFromLabelIndexStorage(value_str)};
-  return CreateVertex(index_accessor, gid, labels, utils::DeserializePropertiesFromLabelIndexStorage(value_str),
-                      index_delta);
+  std::vector<LabelId> labels_id{utils::DeserializeLabelsFromLabelIndexStorage(value)};
+  PropertyStore properties{utils::DeserializePropertiesFromLabelIndexStorage(value)};
+  return CreateVertex(index_accessor, gid, labels_id, std::move(properties), index_delta);
 }
 
 std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToLabelPropertyIndexCache(
-    const rocksdb::Slice &key, const rocksdb::Slice &value, Delta *index_delta,
+    std::string &&key, std::string &&value, Delta *index_delta,
```

Comment:
Why not use `const &` here on key?

Reply:
And `value` too

Reply:
because I think it is better like this, I don't need this value anymore in the caller so I just "steal" them

Reply:
If I am not wrong it should work correctly even if you pass `const &` due to reference collapsing, so on that side I think we are good. To be honest I am not just sure if it is the correct design pattern, what are flows in approach if you only function only accepts && ref. I think "having it as it is now" can maybe shoot us in the leg afterward.

If I remember correctly from a book I read, I think && ref should be used only to signal that you plan to move it to a different variable and in case of perfect forwarding when you want to pass const & or &&.

Reply:
yes true and in this case I don't want to move it but just use it in the function and destroy it. And about reference collapsing, it will work yes but this should be more efficient. Because you basically discard it